### ISSUE TITLE
Multiple local scope usage fails when using Builder->when()

### ISSUE THREAD

**Issue by spaceemotion** [nunomaduro/larastan#351]
- Larastan Version: 0.4.3
- `--level` used: max

### Description
Referencing a single scoped method (e.g. `$query->isActive()`) works, but chaining breaks if it's inside a closure that's used by `$query->when(true, $closure)`:

>  Call to an undefined method Illuminate\Database\Eloquent\Collection&iterable<int, Illuminate\Database\Eloquent\Builder>::released().

### Laravel code where the issue was found

```php
class Model {
  public function scopeIsActive($query) {
    $query->where('is_active', true);
  }

  public function scopeReleased($query) {
    $query->whereNotNull('released_at');
  }
}


Model::query()->when(true, function (Builder $query) {
  $query->isActive()->released();
});
```

**Comment by spaceemotion** [comment]
Just noticed a similar behavior when using this with relationship queries:

```php
class User
{
    // this works
    public function draftContract(): HasOne
    {
        return $this->hasOne(Contract::class)
            ->whereNull('accepted_at')
            ->latest();
    }

    // this does not
    public function activeContract(): HasOne
    {
        return $this->hasOne(Contract::class)
            ->whereNotNull('accepted_at')
            ->whereNotNull('released_at')
            ->latest();
    }
}
```

gives me the following:

> Method App\Models\User::activeContract() should return Illuminate\Database\Eloquent\Relations\HasOne but returns Illuminate\Database\Query\Builder.

It looks like the context does not get transferred over when there's more than one "level" to it.

**Comment by spaceemotion** [comment]
Might be related to https://github.com/nunomaduro/larastan/issues/352#issuecomment-547306680

**Comment by canvural** [comment]
Hi,

For the first example; with 0.5.1, builders are now generics. So you can help Larastan with an annotation like `/** @var Builder<Model> $query */` Maybe later we can solve this in the code with the help of generics.

For your second comment, it should be fixed in 0.5.1 So please test again and let me know :+1:

**Comment by canvural** [comment]
As I said before, this should be fixed. I'm gonna close this issue because of the inactivity. Please open a new issue if you still have the same error with the new versions. Thank you.
